### PR TITLE
Feature/52058 - FileUpload

### PR DESCRIPTION
**Pull request by Aekal**
dev-52058

### CODE REVIEW

File: components/src/molecules/file-upload/FileUpload.scss [hunk 1]
```diff
@@ -4,6 +4,7 @@ $file-upload__margin        : $spacer--medium !default;
 $file-upload__field-height  : 48px !default;
 $file-upload__font-size     : 14px !default;
```
Comment (szafran89) [inline components/src/molecules/file-upload/FileUpload.scss]: can we use variable?

File: components/src/molecules/file-upload/FileUpload.stories.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+storiesOf('Molecules/File Upload', module)
+  .addParameters({ info: true })
+  .add(
+    'Default',
+    () => ({
+      components: { AFileUpload },
+      template: `
+        <a-file-upload
+          accepted-formats=".pdf,.doc"
+        >
```
Comment (szafran89) [inline components/src/molecules/file-upload/FileUpload.stories.js]: if one attr we can keep in one line

File: components/src/molecules/file-upload/FileUpload.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  >
+    <span class="file-upload__label">
+      <!-- @slot Top title content -->
+      <slot name="title"/>
+    </span>
+
+    <a-button
+      class="button file-upload__button"
+      aria-controls="file"
+      @click="chooseFile"
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: You're right, added, thanks for spotting that :)

File: components/src/molecules/file-upload/FileUpload.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<div class="file-upload">
+  <input
+    :class="['file-upload__input', inputClass]"
+    :id="inputId"
+    :name="inputName"
+    type="file"
+    ref="fileInput"
+    :accept="acceptedFormats"
+    @change="updateFile"
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: Fixed

File: components/src/molecules/file-upload/FileUpload.html [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  >
+  <label
+    :class="['file-upload__wrapper', labelClass]"
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: Done

File: components/src/molecules/file-upload/FileUpload.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    :for="inputId"
+  >
+    <span class="file-upload__label">
+      <!-- @slot Top title content -->
+      <slot name="title"/>
+    </span>
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: Done

File: components/src/molecules/file-upload/FileUpload.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    <a-button
+      class="button file-upload__button"
+      aria-controls="file"
+      @click="chooseFile"
+    >
+      <!-- @slot Upload button text content -->
+      <slot name="button"/>
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: Done

File: components/src/molecules/file-upload/FileUpload.html [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    <span
+      ref="fileName"
+      class="file-upload__name"
+      tabindex="0"
+    >
+      <span v-if="fileName && !error">
+        {{ fileName }}
+      </span>
+      <span v-else-if="error" class="file-upload__error">
+        <!-- @slot Error message - displayed when file is too big -->
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: Done

File: components/src/molecules/file-upload/FileUpload.html [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    </a-button>
+
+    <span
+      ref="fileName"
+      class="file-upload__name"
+      tabindex="0"
+    >
+      <span v-if="fileName && !error">
+        {{ fileName }}
+      </span>
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: I used scoped slot for this

File: components/src/molecules/file-upload/FileUpload.html [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    >
+      <!-- @slot Upload button text content -->
+      <slot name="button"/>
+    </a-button>
+
+    <span
+      ref="fileName"
+      class="file-upload__name"
+      tabindex="0"
+    >
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: I'm not sure if it's possible to use nested slots, I didn't found any good solution, do you have any suggestion how to do it?

**Comment by Aekal** [comment]
I changed slots to our new approach, but after adding default text values to slots now we have 4 new props: `title`, `buttonText`, `errorText`, `emptyText`. Do we want to keep them there or remove and replace with static text in `.html` instead? We can always use slot to update default text, what do think? Please take a look at current FileUpload implementation and let me know if there's still something to improve :)

**Comment by szafran89** [comment]
IMO we should keep them, the same approach as here from banner component https://github.com/DivanteLtd/storefront-ui/tree/master/src/components/molecules/SfBanner

**Comment by szafran89** [comment]
For now when I click here http://prntscr.com/n9fyc6 there is shown file browse area. Should be visible only after clik `Upload a file`

File: components/src/molecules/file-upload/FileUpload.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    title: {
+      type: String,
+      default: 'Your file'
+    },
+    /**
+     * Button text
+     */
+    buttonText: {
+      type: String,
+      default: 'Upload a file'
```
Comment (szafran89) [inline components/src/molecules/file-upload/FileUpload.js]: we can't pass default values for string props b/c translations, move to the story, please change for each occurs

File: components/src/molecules/file-upload/FileUpload.html [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    :for="inputId"
+  >
+    <!-- @slot Top title content (Named slot) -->
+    <slot name="title">
+      <span class="file-upload__label">
+        {{ title }}
+      </span>
+    </slot>
+
+    <a-button
```
Comment (szafran89) [inline components/src/molecules/file-upload/FileUpload.html]: move a button inside `button` slot and add condition `v-if="button"`

File: components/src/molecules/file-upload/FileUpload.html [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  <input
+    :class="['file-upload__input', inputClass]"
+    :id="inputId"
+    :name="inputName"
+    type="file"
+    ref="fileInput"
+    :accept="acceptedFormats"
+    @change="updateFile"
+  >
+  <a-label
```
Comment (szafran89) [inline components/src/molecules/file-upload/FileUpload.html]: I'm wondering if it's correct to wrap whole content to label tag?

File: components/src/molecules/file-upload/FileUpload.html [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  >
+    <!-- @slot Top title content (Named slot) -->
+    <slot name="title">
+      <span class="file-upload__label">
+        {{ title }}
+      </span>
+    </slot>
+
+    <a-button
+      class="button file-upload__button"
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: Removed

File: components/src/molecules/file-upload/FileUpload.js (hunk 11 above)
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.js]: Ok, done

File: components/src/molecules/file-upload/FileUpload.html (hunk 13 above)
Comment (Aekal) [reply components/src/molecules/file-upload/FileUpload.html]: I think it's ok to use label as a wrapper to some texts and a button

Comment (szafran89) [reply components/src/molecules/file-upload/FileUpload.html]: let's keep this shape for now

**Pull request merged**